src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSECSResourceDetector.cs

```diff
@@ -31,42 +37,140 @@ public class AWSECSResourceDetector : IResourceDetector
     /// <summary>
     /// Detector the required and optional resource attributes from AWS ECS.
     /// </summary>
-    /// <returns>List of key-value pairs of resource attributes.</returns>
-    public IEnumerable<KeyValuePair<string, object>> Detect()
+    /// <returns>Resource.</returns>
+    public Resource Detect()
     {
-        List<KeyValuePair<string, object>> resourceAttributes = null;
-
         if (!this.IsECSProcess())
         {
-            return resourceAttributes;
+            return null;
         }
 
+        var resourceAttributes = new List<KeyValuePair<string, object>>()
+        {
+            new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudProvider, "aws"),
+            new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudPlatform, "aws_ecs"),
+        };
+
         try
         {
             var containerId = this.GetECSContainerId(AWSECSMetadataPath);
+            resourceAttributes.AddRange(ExtractResourceAttributes(containerId));
+        }
+        catch (Exception ex)
+        {
+            AWSXRayEventSource.Log.ResourceAttributesExtractException(nameof(AWSECSResourceDetector), ex);
+        }
 
-            resourceAttributes = this.ExtractResourceAttributes(containerId);
+        try
+        {
+            resourceAttributes.AddRange(ExtractMetadataV4ResourceAttributes());
         }
         catch (Exception ex)
         {
             AWSXRayEventSource.Log.ResourceAttributesExtractException(nameof(AWSECSResourceDetector), ex);
         }
 
-        return resourceAttributes;
+        return new Resource(resourceAttributes);
+    }
+
+    internal static string GetStringOrThrow(JObject obj, string fieldName)
+    {
+        var value = obj[fieldName]?.ToString();
+        if (value == null)
+        {
+            throw new ArgumentNullException($"The expected '{fieldName}' field is missing");
```

Comment:
I think you have a point about the access to the `LogDriver` field: if no logging is required, as far as I can tell, it will be unspecified. The other fields accessed via `GetStringOrThrow` are part of the ECS Metadata v4 contract as I understand it.

Reply:
Done in f35d9eb